### PR TITLE
Handle empty text with no line metrics in RenderEditable._lineNumberFor

### PR DESCRIPTION
**Pull request by jason-simmons**
SkParagraph will return an empty list of line metrics if the text
is empty.

### CODE REVIEW

**Comment by jason-simmons** [comment]
This is tested by the "Arrow keys don't move input focus" tests in `test/material/text_field_test.dart` and `test/cupertino/text_field_test.dart` when running with SkParagraph.

**Review by LongCatIsLooong — COMMENTED** [review]
Thanks for the fix. Would it be possible to add a test or do we have to request test exemption for this one? Does libtxt not return an empty list if there's no text?

File: packages/flutter/lib/src/rendering/editable.dart [hunk 1]
```diff
@@ -2404,6 +2404,9 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
   MapEntry<int, Offset> _lineNumberFor(TextPosition startPosition, List<ui.LineMetrics> metrics) {
     // TODO(LongCatIsLooong): include line boundaries information in
     // ui.LineMetrics, then we can get rid of this.
+    if (metrics.isEmpty && startPosition.offset == 0) {
+      return MapEntry<int, Offset>(0, Offset(0,0));
+    }
```
[the remark this answers is not in the document]
Comment (jason-simmons) [reply packages/flutter/lib/src/rendering/editable.dart]: done

**Comment by jason-simmons** [comment]
Unlike SkParagraph, in this situation Libtxt returns a nonempty list containing a line with height but no width.  I think either result meets the specification of the API.

**Review by LongCatIsLooong — APPROVED** [review]

**Pull request merged**